core/core.go

```diff
@@ -209,12 +209,14 @@ func (c *Core) Stop() {
 
 // WriteBlock write the block to the bodydb database
 func (c *Core) WriteBlock(block *types.Block) {
-	isDomCoincident := c.sl.engine.IsDomCoincident(block.Header())
-	// Only add non dom blocks to the append queue
-	if !isDomCoincident {
-		c.addToAppendQueue(block)
+	if c.GetBlockByHash(block.Hash()) == nil {
+		isDomCoincident := c.sl.engine.IsDomCoincident(block.Header())
```

Comment:
Should be a strict order check, no?

Reply:
we don't have order function yet, this will change in POEM